src/figuresClasses.ts

```diff
@@ -1,19 +1,80 @@
+type Shape = ('triangle' | 'circle' | 'rectangle');
+type Color = ('red' | 'green' | 'blue');
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 && this.b <= 0 && this.c <= 0) {
+      throw new Error('Size must be greater than 0');
+    }
+
```

Comment:
will be better to create separate function that will accept input arguments and will check that each of them is bigger than 0
and call this function for each figure